param/coercer/coercer.go

```diff
@@ -122,40 +101,71 @@ func coercePrimitiveType(val interface{}, primitiveType string) (interface{}, bo
 			return nil, false
 		}
 		return valFloat, true
+
+	case primitiveType == stringType:
+		return valStr, true
 	}
 
 	return nil, false
 }
 
-// coerceSchema tries to coerce a schema containing a primitive type from the
-// given generic interface{} value.
+// coerceNonObjectSchema tries to coerce a non-object schema given generic interface{} value.
 //
 // It's similar to coercePrimitiveType above (and indeed calls into it), but
-// also handles the case of an anyOf schema that supports a number of different
-// primitve types.
-func coerceSchema(val interface{}, schema *spec.Schema) (interface{}, bool) {
+// also handles array and anyOf schema (supporting a number of different primitive types)
+func coerceNonObjectSchema(val interface{}, schema *spec.Schema) (interface{}, bool, error) {
```

Comment:
`anyOf` and `array` block below will recurse its sub-schema with `coerceSubSchema`